core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -333,6 +351,17 @@ public BlockOutStream getOutStream(long blockId, long blockSize, OutStreamOption
           "Not enough workers for replications, %d workers selected but %d required",
           workerAddressList.size(), initialReplicas));
     }
+
+    // ALLUXIO-11172: If the worker is in a container, use the container hostname
+    // to establish the connection.
+    for (WorkerNetAddress workerAddress : workerAddressList) {
```

Comment:
same here, this block also ends up calling `NetworkAddressUtils` to create `GrpcDataWriter`